core/src/test/java/com/linecorp/armeria/client/encoding/TestStreamDecoder.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.encoding;
+
+import com.linecorp.armeria.common.ContentTooLargeException;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.UnexpectedDecodeException;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.encoding.StreamDecoder;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.channel.embedded.EmbeddedChannel;
+import io.netty.handler.codec.compression.DecompressionException;
+
+class TestStreamDecoder implements StreamDecoder {
```

Comment:
What do you think of extending `AbstractStreamDecoder` instead of copying the code?
If necessary, we may move this class to `common.encoding`.

Reply:
That would be great. I've changed this class' directory path. Extending AbstractStreamDecoder for the test class is what i wanted, not recreating the same code. Thanks for allowed to do this.